### PR TITLE
Publicly expose the currently selected link and allow setting initially selected link

### PR DESCRIPTION
**Pull request by cmalonzo**
**Scenario solving for:** In the FilePicker component, we have an instance of a Nav component on the left part and on the right are views of different sources from which a user can select a file. In this way, we're using the Nav to "navigate" between the different views on the right and this requires the FilePicker to know which Nav item has been clicked. Right now, we're traversing the DOM to find which child in the FilePicker's Nav component has the class "is-selected" and this isn't the ideal solution for various reasons, one glaring one: this logic in the filepicker tightly couples it to and is necessarily vulnerable to classname changes in fabric-react. 

**Proposed solution:** publicly expose the key of currently selected nav item. This would only work if all links in the group(s) of links passed to a Nav instance have populated and individually unique 'key' property.

If this is the approved solution, I can write up documentation in the demo pages to accompany this change.

### CODE REVIEW

File: src/components/Nav/Nav.Props.ts [hunk 1]
```diff
@@ -21,6 +21,11 @@ export interface INavProps {
    * Indicates whether the navigation component renders on top of other content in the UI
    */
   isOnTop?: boolean;
+
+  /**
+   * (Optional) The key of the nav item initially selected.
+   */
+  initialSelectedKey?: string;
```
Comment (dzearing) [inline src/components/Nav/Nav.Props.ts]: What if you called this defaultSelectedKey to be consistent with other uncontrolled properties? (e.g. defaultChecked)

File: src/components/Nav/Nav.tsx [hunk 2]
```diff
@@ -43,12 +49,22 @@ export class Nav extends React.Component<INavProps, INavState> {
...
   }
 
+  public get selectedKey(): string {
+    return this._selectedKey;
+  }
+
   private _renderLink(link: INavLink, linkIndex: number): React.ReactElement<{}> {
     let { onLinkClick } = this.props;
+
+    const ifLinkSelected: boolean = _isLinkSelected(link, this._selectedKey);
```
Comment (antonlabunets) [inline src/components/Nav/Nav.tsx]: isLinkSelected

File: src/components/Nav/Nav.tsx [hunk 3]
```diff
@@ -43,12 +49,22 @@ export class Nav extends React.Component<INavProps, INavState> {
...
+    if (ifLinkSelected) {
+      this._selectedKey = link.key ? link.key : undefined;
+    }
+
     return (
       <li key={ linkIndex }>
         <a
-          className={'ms-Nav-link' + (_isLinkSelected(link) ? ' is-selected' : '')}
+          className={'ms-Nav-link' + (ifLinkSelected ? ' is-selected' : '')}
```
Comment (antonlabunets) [inline src/components/Nav/Nav.tsx]: Minor: can we use the css() helper here?

**Comment by antonlabunets** [comment]
It would be good to update the example page with your suggested usage. I guess the only thing I would wish you is to make onLinkSelected send out the key in params inside of having to do one more request after. Otherwise this certainly fits our case.

**Comment by inateeg** [comment]
I don't understand this comment since example is added?

Sent from my iPhone

On Aug 12, 2016, at 11:08 PM, antonlabunets <notifications@github.com<mailto:notifications@github.com>> wrote:


It would be good to update the example page with your suggested usage. I guess the only thing I would wish you is to make onLinkSelected send out the key in params inside of having to do one more request after. Otherwise this certainly fits our case.

-

**Comment by antonlabunets** [comment]
I meant updating our example page for Nav with intended usage.

**Comment by inateeg** [comment]
I have added usage in example of ‘target’  and ‘key’ is meta info for support editing no view UI change?  onLinkSelected already there if you see test page leftNav selected state?

File: src/components/Nav/Nav.tsx [hunk 4]
```diff
@@ -115,13 +131,17 @@ export class Nav extends React.Component<INavProps, INavState> {
...
 
-function _isLinkSelected(link: INavLink): boolean {
+function _isLinkSelected(link: INavLink, selectedKey: string): boolean {
     if (!link.url) {
       return false;
     }
     _urlResolver.href = link.url || '';
     const target: string = _urlResolver.href;
 
+    if (selectedKey && link.key && link.key === selectedKey) {
```
Comment (MBSilverstein) [inline src/components/Nav/Nav.tsx]: You can simplify this condition to (selectedKey && link.key === selectedKey).

**Comment by cmalonzo** [comment]
Updated with notes on how to use selectedKey, refactored to use css(), and suggested name changes.

**Comment by antonlabunets** [comment]
Approved from picker perspective

**Pull request merged**